crates/nixci/src/command/core.rs

```diff
@@ -39,7 +39,10 @@ impl Command {
         tracing::info!("{}", "\n👟 Reading om.ci config from flake".bold());
         let cfg = self.get_config(nixcmd).await?;
         match self {
-            Command::Run(cmd) => cmd.run(nixcmd, verbose, cfg).await,
+            Command::Run(cmd) => match &cmd.steps_args.build_step_args.on {
+                Some(host) => cmd.run_remote(nixcmd, cfg, host).await,
+                None => cmd.run(nixcmd, verbose, cfg).await,
+            },
```

Comment:
The `run` command is responsible for ... well ... all that `om ci run` does. Ergo, remote run logic should not spill over here; rather it should remain encapsulated in `command/run.rs` (you can still keep that `run_remote` function but as private function to that module).